lib/app.dart

```diff
@@ -35,4 +44,44 @@ class App {
         break;
     }
   }
+
+  void initFirebase() async {
+    await fb.init('firebase/constants.json');
+    utils.setLoginDomains(fb.allowedEmailDomains);
+    fb.firebaseAuth.onAuthStateChanged.listen(_fbAuthChanged);
+    loginButton.onClick.listen((_) {
+      utils.disableLoginButton();
+      fb.signInWithGoogle();
+    });
+  }
+
+  void _fbAuthChanged(firebase.User user) async {
+    utils.enableLoginButton();
+    if (user == null) {
+      logger.debug('User not signed in');
+      utils.showLoginModal();
+      return;
+    }
+
+    if (!fb.allowedEmailDomains.any((domain) => user.email.endsWith(domain))) {
+      logger.error('Email domain not allowed');
+      await fb.deleteUser();
+      utils.showLoginError('Email domain not allowed');
```

Comment:
This and the other `utils.` in this file don't feel like utils-type functions to me. Consider having them in a view instead?

Reply:
I have moved this into a view under login - more details in the comment further below